Documentation/templates/files/swift5.md

```diff
@@ -0,0 +1,51 @@
+## Template Information
+
+| Name      | Description       |
+| --------- | ----------------- |
+| File name | files/swift5.stencil |
+| Configuration example | <pre>files:<br />  inputs: dir/to/search/with/filter<br />  filter: .+\.*$<br />  outputs:<br />    templateName: swift5<br />    output:Files.swift</pre> |
+| Language | Swift 5 |
+| Author | Mike Gray |
+
+## When to use it
+
+- When you need to generate *Swift 5* code.
+
+## Customization
+
+| Parameter Name | Default Value | Description |
+| -------------- | ------------- | ----------- |
+| `bundle` | `BundleToken.bundle` | Allows you to set from which bundle files are loaded from. By default, it'll point to the same bundle as where the generated code is. |
+| `enumName` | `Files` | Allows you to change the name of the generated `enum` containing the file tree. |
+| `resourceTypeName` | `File` | Allows you to change the name of the struct type representing a file. |
+| `publicAccess` | N/A | If set, the generated constants will be marked as `public`. Otherwise, they'll be declared `internal`. |
+
+## Generated Code
+
+**Extract:**
+
+```swift
+internal enum Files {
+  /// test.txt
+  internal static let testTxt = File(name: "test", ext: "txt", path: "", mimeType: "text/plain")
+  /// subdir/
+  internal enum subdir {
+    /// subdir/A Video With Spaces.mp4
+    internal static let aVideoWithSpacesMp4 = File(name: "A Video With Spaces", ext: "mp4", path: "subdir", mimeType: "video/mp4")
+  }
+}
+```
+
+[Full generated code](../../../Tests/Fixtures/Generated/Files/swift5/defaults.swift)
+
+## Usage example
+
+```swift
+// Access files using the `url` or `path` fields
+let txt = Files.testTxt.url
+let video = Files.subdir.aVideoWithSpacesMp4.path
```

Comment:
Looking at the example, I'm wondering if it wouldn't be nice to have a template parameter `useExtension` (boolean) or similar (not super inspired for the name) that would allow to not use the extension in the generated Swift constant name. i.e. that would turn this line into `Files.subdir.aVideoWithSpaces.path`?

_Of course, that would generate a risk of conflicts (if there are 2 files with the same basename but only their extension being different), but it's not much different than the case of the flat template; I think many people won't have conflict and will thus prefer disabling using the extension in the constant's name, and people having conflicting file names will want the extension to avoid conflicts_

Reply:
Would definitely be useful for this case/parser. Collision issues are up to the user.

What the default should be (on/off) is another matter 😅 It's safer with extensions (less collisions), but without gives nicer enums.

Reply:
That's a pretty good idea. I'll add it